pkg/collector/remoteconfig/remote_config.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package remoteconfig schedule and unschedule integrations through remote-config
+// using the AGENT_INTEGRATIONS product
+package remoteconfig
+
+import (
+	"encoding/json"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/collector"
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+	"github.com/DataDog/datadog-agent/pkg/secrets"
+	pkglog "github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// Scheduler is the structure used to run checks with RC
+type Scheduler struct {
+	scheduler     *collector.CheckScheduler
+	runningChecks []integration.Config
+}
+
+type agentIntegration struct {
+	Name       string            `json:"name"`
+	Instances  []json.RawMessage `json:"instances"`
+	InitConfig json.RawMessage   `json:"init_config"`
+}
+
+// secretsDecrypt allows tests to intercept calls to secrets.Decrypt.
+var secretsDecrypt = secrets.Decrypt
+
+// NewScheduler creates an instance of a remote config integration scheduler
+func NewScheduler() *Scheduler {
+	return &Scheduler{
+		runningChecks: make([]integration.Config, 0),
+	}
+}
+
+// AddScheduler initiate the remote-config scheduler
+func (sc *Scheduler) AddScheduler(scheduler *collector.CheckScheduler) {
```

Comment:
Is there ever a use case for creating a `remoteconfig.Scheduler` without having a `collector.CheckScheduler` already created and ready to use as the scheduler source? If not, this should be in the constructor for a cleaner API.